StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1023DereferenceAndAccessOfSymbolsMustBeSpacedCorrectly.cs

```diff
@@ -121,22 +122,10 @@ private void HandleAsteriskToken(SyntaxTreeAnalysisContext context, SyntaxToken
                 return;
             }
 
-            bool precededBySpace;
-            bool firstInLine;
-
-            firstInLine = token.HasLeadingTrivia || token.GetLocation()?.GetMappedLineSpan().StartLinePosition.Character == 0;
-            if (firstInLine)
-            {
-                precededBySpace = true;
-            }
-            else
-            {
-                SyntaxToken precedingToken = token.GetPreviousToken();
-                precededBySpace = precedingToken.HasTrailingTrivia;
-            }
-
-            bool followedBySpace = token.HasTrailingTrivia;
-            bool lastInLine = followedBySpace && token.TrailingTrivia.Any(SyntaxKind.EndOfLineTrivia);
+            bool firstInLine = token.IsFirstInLine();
+            bool precededBySpace = firstInLine ? true : token.IsPrecededBySpace();
```

Comment:
You could also write this like `precededBySpace = firstInLine || token.IsPrecededBySpace();`